File: client/android/app/src/main/kotlin/vn/zuni/findpurpose/models/Question.kt
package vn.zuni.findpurpose.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import vn.zuni.findpurpose.extensions.emptyString
import java.io.Serializable

data class Question constructor(@SerializedName("index") @Expose var index: Int = 0,
                                @SerializedName("question_type") @Expose var questionType: String? = emptyString,
                                @SerializedName("question_title") @Expose var questionTitle: String? = emptyString,
                                @SerializedName("description") @Expose var description: String? = emptyString,
                                @SerializedName("required") @Expose var required: Boolean = false,
                                @SerializedName("random_choices") @Expose var randomChoices: Boolean = false,
                                @SerializedName("choices") @Expose var choices: List<String> = emptyList(),
                                @SerializedName("min") @Expose var min: Int = 0,
                                @SerializedName("max") @Expose var max: Int = 0,
                                @SerializedName("number_of_lines") @Expose var numberOfLines: Int = 0)
    : Serializable

File: client/ios/eSurvey/eSurvey/models/Question.swift
//
//  Question.swift
//  eSurvey
//
//  Created by Nam Nguyen on 1/13/18.
//  Copyright © 2018 Unilever. All rights reserved.
//
import ObjectMapper

open class Question: Mappable {
    
    open var index: Int = 0
    open var questionType: String!
    open var questionTitle: String!
    open var description: String!
    open var required: Bool = false
    open var randomChoices: Bool = false
    open var choices : [String] = [String]()
    open var min: Int = 0
    open var max: Int = 0
    open var numberOfLines: Int = 0
    
    public init() {
    }
    
    required public init?(map: Map){
    }
    
    open func mapping(map: Map) {
        index <- map["index"]
        questionType <- map["question_type"]
        questionTitle <- map["question_title"]
        description <- map["description"]
        required <- map["required"]
        randomChoices <- map["random_choices"]
        choices <- map["choices"]
        min <- map["min"]
        max <- map["max"]
        numberOfLines <- map["number_of_lines"]
    }
}
